Sources/ClientRuntime/Networking/Http/Middlewares/FlexibleChecksumsResponseMiddleware.swift

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
+
+public struct FlexibleChecksumsResponseMiddleware<OperationStackOutput>: Middleware {
+    
+    public let id: String = "FlexibleChecksumsResponseMiddleware"
+    
+    // The priority to validate response checksums, if multiple are present
+    let CHECKSUM_HEADER_VALIDATION_PRIORITY_LIST: [String] = [
+        HashFunction.crc32c,
+        .crc32,
+        .sha1,
+        .sha256
+    ].map { $0.toString() }
+
+    let validationMode: Bool
+
+    public init(validationMode: Bool) {
+        self.validationMode = validationMode
+    }
+
+    public func handle<H>(context: Context,
+                          input: SdkHttpRequest,
+                          next: H) async throws -> OperationOutput<OperationStackOutput>
+    where H: Handler,
+    Self.MInput == H.Input,
+    Self.MOutput == H.Output,
+    Self.Context == H.Context {
+
+        // The name of the checksum header which was validated. If `null`, validation was not performed.
+        context.attributes.set(key: AttributeKey<String>(name: "ChecksumHeaderValidated"), value: nil)
+
+        // Initialize logger
+        guard let logger = context.getLogger() else {
+            throw ClientError.unknownError("No logger found!")
+        }
+        
+        // Exit if validation should not be performed
+        if (!validationMode) {
+            logger.info("Checksum validation should not be performed! Skipping workflow...")
+            return try await next.handle(context: context, input: input)
+        }
+
+        // Get the response
+        let response = try await next.handle(context: context, input: input)
+        
+        // Determine if any checksum headers are present
+        logger.debug("HEADERS: \(response.httpResponse.headers)")
+        let _checksumHeader = CHECKSUM_HEADER_VALIDATION_PRIORITY_LIST.first { response.httpResponse.headers.value(for: "x-amz-checksum-\($0)") != nil }
```

Comment:
Underscores are usually reserved for private variables in Swift.  I think you can just drop the underscore

Reply:
I can make this variable private since the variable that is actually utilized elsewhere is the guard statement